pkg/cluster/nodes/nodes.go

```diff
@@ -134,3 +134,49 @@ func WaitForReady(node *Node, until time.Time) bool {
 		return true
 	})
 }
+
+// Restart restarts nodes by name / ID (see Node.String())
+func Restart(nodes []Node) error {
+	if len(nodes) == 0 {
+		return nil
+	}
+	ids := []string{}
+	for _, node := range nodes {
+		ids = append(ids, node.name)
+	}
+	cmd := exec.Command(
+		"docker",
+		append(
+			[]string{
+				"restart",
+				// the default wait time is 10s
+				"--time=1",
+			},
+			ids...,
+		)...,
+	)
+
+	if err := cmd.Run(); err != nil {
+		// TODO: logging here
+		return err
+	}
+
+	for _, node := range nodes {
+
+		if err := node.FixMounts(); err != nil {
+			// TODO: logging here
+			return err
+		}
+
+		if err := node.SignalStart(); err != nil {
+			// TODO: logging here
+			return err
+		}
+
+		if !node.WaitForDocker(time.Now().Add(time.Second * 60)) {
```

Comment:
i think this PR will be affected by the changes in:
https://github.com/kubernetes-sigs/kind/pull/461

Reply:
yes. after the PR has been merged these codes remove.

Reply:
I need to rework HAproxy in the new world :tm: as well, so those parts will be getting a PR soon and also will need updating... sorry about that!

Reply:
do not mind. as long as we can push things forward, it will suffice.